specs/src/group.rs

```diff
@@ -0,0 +1,150 @@
+
+use World;
+
+#[cfg(feature="serialize")]
+use Entity;
+#[cfg(feature="serialize")]
+use serde::{self, Serializer, Deserializer};
+
+/// Group of components. Can be subgrouped into other component groups.
+pub trait ComponentGroup {
+    /// Components defined in this group, not a subgroup.
+    fn local_components() -> Vec<&'static str>;
+    /// Components defined in this group along with subgroups.
+    fn components() -> Vec<&'static str>;
+    /// Subgroups included in this group.
+    fn subgroups() -> Vec<&'static str>;
+    /// Registers the components into the world.
+    fn register(&mut World);
+}
+
+/// Group of serializable components.
+#[cfg(feature="serialize")]
+pub trait SerializeGroup: ComponentGroup {
+    /// Serializes the group of components from the world.
+    fn serialize_group<S: Serializer>(world: &World, serializer: S) -> Result<S::Ok, S::Error>;
+    /// Helper method for serializing the world.
```

Comment:
Do we need to expose these helpers in the trait or are they only called in trait implementation? If they are not needed anywhere else then these helpers are just implementation detail and should probably removed.